src/Analyzers/CSharp/CodeFixes/UseIsNullCheck/CSharpUseIsNullCheckForReferenceEqualsCodeFixProvider.cs

```diff
@@ -26,33 +29,46 @@ protected override string GetIsNullTitle()
         protected override string GetIsNotNullTitle()
             => GetIsNullTitle();
 
-        private static SyntaxNode CreateEqualsNullCheck(SyntaxNode argument)
-            => SyntaxFactory.BinaryExpression(
-                SyntaxKind.EqualsExpression,
-                (ExpressionSyntax)argument,
-                SyntaxFactory.LiteralExpression(SyntaxKind.NullLiteralExpression)).Parenthesize();
+        private static readonly LiteralExpressionSyntax s_nullLiteralExpression
+            = LiteralExpression(SyntaxKind.NullLiteralExpression);
+
+        private static readonly ConstantPatternSyntax s_nullLiteralPattern
+            = ConstantPattern(s_nullLiteralExpression);
+
+        private static SyntaxNode CreateEqualsNullCheck(ExpressionSyntax argument)
+            => BinaryExpression(SyntaxKind.EqualsExpression, argument, s_nullLiteralExpression).Parenthesize();
 
-        private static SyntaxNode CreateIsNullCheck(SyntaxNode argument)
-            => SyntaxFactory.IsPatternExpression(
-                (ExpressionSyntax)argument,
-                SyntaxFactory.ConstantPattern(SyntaxFactory.LiteralExpression(SyntaxKind.NullLiteralExpression))).Parenthesize();
+        private static SyntaxNode CreateIsNullCheck(ExpressionSyntax argument)
+            => IsPatternExpression(argument, s_nullLiteralPattern).Parenthesize();
 
-        private static SyntaxNode CreateIsNotNullCheck(SyntaxNode argument)
+        private static SyntaxNode CreateIsNotNullCheck(ExpressionSyntax argument)
         {
-            return SyntaxFactory
-                .BinaryExpression(
-                    SyntaxKind.IsExpression,
-                    (ExpressionSyntax)argument,
-                    SyntaxFactory.PredefinedType(SyntaxFactory.Token(SyntaxKind.ObjectKeyword)))
-                .Parenthesize();
+            var parseOptions = (CSharpParseOptions)argument.SyntaxTree.Options;
+
+            // TODO: this needs to be CSharp9 once that version is available.
+#if !CODE_STYLE
+            if (parseOptions.LanguageVersion >= LanguageVersion.CSharp8)
```

Comment:
note: this is wrong.  discussing with @gafter getting an actual CSharp9 lang version that we can test against.